sdcm/cluster.py

```diff
@@ -2643,10 +2643,10 @@ def get_nodes_status(self):
         nodes_status = {}
         try:
             statuses = self.parent_cluster.get_nodetool_status(verification_node=self)
-
+            node_ip_map = {ip: node for node in self.parent_cluster.nodes for ip in node.get_all_ip_addresses()}
             for dc, dc_status in statuses.items():
                 for node_ip, node_properties in dc_status.items():
-                    if node := self.parent_cluster.find_node_by_ip(node_ip):
+                    if node := node_ip_map.get(node_ip):
```

Comment:
> There are also `get_peers_info` and `get_gossip_info`, which, moreover, are wrapped with retries decorator.

these methods fail in case ip is not found and retry - if we cache it will be failing until cache is deleted (or ttl'ed) making it unusable. 

I'll reuse current solution there too.